### PR TITLE
test: compare source and qgis path strokes

### PR DESCRIPTION
**Pull request by ebelo**
## Summary
- add source-vs-QGIS visible stroke-control comparisons to the path/pedestrian focus report
- match QGIS-generated layer variants back to their original Mapbox style-layer ids before comparing stroke controls
- include markdown output and tests for exact, split variant, and unmatched stroke-control rows

Refs #949.

## Validation
- `python3 -m pytest tests/test_mapbox_outdoors_path_pedestrian_focus.py -q --tb=short`
- `python3 -m pytest tests/ -x -q --tb=short`
- `python3 scripts/package_plugin.py`
- `python3 validation/mapbox_outdoors_path_pedestrian_focus.py --road-features-json debug/mapbox-outdoors-road-features/all-cameras/20260520T202841Z/road-features.json --source-style-json debug/mapbox-outdoors-source-style/20260520T042926Z/mapbox-outdoors-v12.json --comparison-summary-json debug/mapbox-outdoors-comparison-post-label-split-live/all-cameras/20260520T234632Z/summary.json` -> `debug/mapbox-outdoors-path-pedestrian-focus/20260521T003814Z/summary.md`

### CODE REVIEW

File: validation/mapbox_outdoors_path_pedestrian_focus.py [hunk 1]
```diff
@@ -965,6 +982,75 @@ def _duplicate_label_diagnostic(camera: Mapping[str, object]) -> dict[str, objec
...
+    for detail in qgis_details:
+        layer_id = str(detail.get("id") or "")
+        source_id = _qfit_base_style_layer_id(layer_id)
+        details_by_source_id[source_id].append(detail)
+    return details_by_source_id
+
+
+def _source_qgis_stroke_control_comparisons(camera: Mapping[str, object]) -> list[dict[str, object]]:
+    source_details = _visible_line_details(camera, "source_path_pedestrian_visible_layer_details")
+    qgis_details = _visible_line_details(camera, "qgis_path_pedestrian_visible_layer_details")
```
[the remark this answers is not in the document]
Comment (ebelo) [reply validation/mapbox_outdoors_path_pedestrian_focus.py:1031]: Addressed in 9339f69 by making the unmatched-row behavior explicit in the Markdown copy and adding a regression test for empty QGIS columns.

File: validation/mapbox_outdoors_path_pedestrian_focus.py [hunk 2]
```diff
@@ -965,6 +972,66 @@ def _duplicate_label_diagnostic(camera: Mapping[str, object]) -> dict[str, objec
...
+    }
+
+
+def _qgis_stroke_details_by_source_id(
+    qgis_details: Iterable[Mapping[str, object]],
+) -> dict[str, list[Mapping[str, object]]]:
+    details_by_source_id: dict[str, list[Mapping[str, object]]] = defaultdict(list)
+    for detail in qgis_details:
+        layer_id = str(detail.get("id") or "")
+        source_id = _qfit_base_style_layer_id(layer_id)
```
[the remark this answers is not in the document]
Comment (ebelo) [reply validation/mapbox_outdoors_path_pedestrian_focus.py:1024]: Addressed in 9f1f6d4 by normalizing road-path split suffixes before grouping QGIS stroke details, with coverage for road-path-z16-plus.

**Comment by ebelo** [comment]
Greptile's deferred-import note is readability-only; leaving the local helper in place because the validation script keeps qfit imports after its script-path setup, and the import is cached after first use.
